packages/react/src/file-input/FileInput.tsx

```diff
@@ -139,83 +138,76 @@ export const FileInput = forwardRef<HTMLInputElement, FileInputProps>(
+				required={required}
 			>
-				{(a11yProps) => {
-					const visibleButtonLabel = `Select file${multiple ? 's' : ''}`;
-					const ariaLabel = [
-						visibleButtonLabel,
-						label,
-						a11yProps['aria-required'] && 'required',
-						a11yProps['aria-invalid'] && 'invalid',
-						getSelectedFilesMessage({ fileNames, multiple }),
-					]
-						.filter(Boolean)
-						.join(', ');
-
-					return (
-						<>
-							<Stack gap={1} justifyContent="flex-start" alignItems="start">
-								<Button
-									{...a11yProps}
-									aria-invalid={undefined} // Buttons not announced as invalid when `aria-invalid`, add announcement to label instead
-									aria-label={ariaLabel}
-									aria-required={undefined} // Buttons can't be `aria-required`, add announcement to label instead
-									disabled={disabled}
-									onBlur={onVisualButtonBlur}
-									onClick={onVisualButtonClick}
-									onFocus={onVisualButtonFocus}
-									ref={visibleButtonRef}
-									size={buttonSize}
-									variant="secondary"
-								>
-									{visibleButtonLabel}
-								</Button>
-
-								<Text
-									breakWords
-									color="muted"
-									{...(fileNames.length && {
-										color: undefined,
-										fontWeight: 'bold',
-									})}
-								>
-									{getSelectedFilesMessage({
-										fileNames,
-										isDisplayed: true,
-										multiple,
-									})}
-								</Text>
-							</Stack>
-
-							<input
-								accept={accept?.toString()}
+				{(a11yProps) => (
+					<>
+						<Stack alignItems="start" gap={0.5} justifyContent="flex-start">
+							<Button
+								aria-describedby={a11yProps['aria-describedby']}
+								aria-label={ariaLabel}
 								disabled={disabled}
-								multiple={multiple}
-								onChange={onChange}
-								type="file"
-								{...props}
-								aria-hidden={true}
-								css={{
-									position: 'absolute',
-									height: 1,
-									width: 1,
-									pointerEvents: 'none',
-								}}
-								ref={hiddenInputRef}
-								tabIndex={-1}
-							/>
-						</>
-					);
-				}}
+								id={a11yProps.id}
+								onBlur={onVisualButtonBlur}
+								onClick={onVisualButtonClick}
+								onFocus={onVisualButtonFocus}
+								ref={visibleButtonRef}
+								size={buttonSize}
+								variant="secondary"
+							>
+								{buttonLabel}
+							</Button>
+
+							<Text
+								breakWords
+								color="muted"
+								{...(fileNames.length && {
+									color: undefined,
+									fontWeight: 'bold',
+								})}
+							>
+								{getSelectedFilesMessage({
+									fileNames,
+									isDisplayed: true,
+									multiple,
+								})}
+							</Text>
+						</Stack>
+
+						<input
+							accept={accept?.toString()}
+							aria-hidden={true}
+							css={visuallyHiddenStyles}
+							disabled={disabled}
+							multiple={multiple}
+							onChange={onChange}
+							ref={hiddenInputRef}
+							tabIndex={-1}
+							type="file"
+							{...props}
```

Comment:
I think this spread should happen earlier. We don't want folks overriding `tabIndex`, `type`, or `aria-hidden`.

Reply:
Huh, it's currently spread last, but yeah, I think you're right

Reply:
I'd changed it in a previous commit but `ref` and `css` probably aren't important:

![image](https://github.com/user-attachments/assets/525d6d0b-145d-4468-bb11-969049ec2af6)